test/distributed/shard_tensor/ops/utils.py

```diff
@@ -0,0 +1,139 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Iterable
+
+import torch
+from torch.distributed.tensor import DTensor, distribute_module
+from torch.distributed.tensor.device_mesh import DeviceMesh
+
+from physicsnemo.distributed import ShardTensor
+
+
+def unparallelize_module(module):
+    """
+    This is the inverse of "distribute_module".  Don't use this except in tests.
+
+    (Why need this?  We're leveraging distribute_module to make sure all
+    ranks have the same weights, if needed, instead of relying on random seeds.)
+    """
+    for name, param in list(module._parameters.items()):
+        if isinstance(param, torch.nn.Parameter) and isinstance(param.data, DTensor):
+            # gather to replicated then unwrap
+            local_tensor = param.data.full_tensor()
+            # replace with a normal Parameter
+            module._parameters[name] = torch.nn.Parameter(
+                local_tensor, requires_grad=param.requires_grad
+            )
+    # recurse into submodules
+    for child in module.children():
+        unparallelize_module(child)
+
+    return module
+
+
+def generate_image_like_data(
+    batch_size: int, C_in: int, spatial_shape: tuple[int, ...]
```

Comment:
Do you want to just allocate on device? Might clean up the test code if you just pass `device` as an argument instead of having to call `.to()`. We can potentially parametrize for `dtype` as well, if that's the case.